File: 0136-single-number/0136-single-number.java
class Solution {
  public int singleNumber(int[] nums) {
    Set<Integer> tally = new HashSet<Integer>();
    for (int num : nums) {
      if (tally.contains(num)) {
        tally.remove(num);
      } else {
        tally.add(num);
      }
    }
    return new ArrayList<Integer>(tally).get(0);
  }
}

File: 0136-single-number/0136-single-number.go
func singleNumber(nums []int) int {
  tally, ans := make(map[int]bool), 0
  for _, num := range nums {
    if _, ok := tally[num]; ok {
      delete(tally, num)
    } else {
      tally[num] = true
    }
  }
  for num := range tally {
    ans = num
    break
  }
  return ans
}